backend_jobs/ingestion_pipeline/pipeline_lib/database_functions.py

```diff
@@ -0,0 +1,143 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+import random
+from datetime import datetime
+import apache_beam as beam
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+from google.cloud import firestore as cloud_firestore
+import geohash2
+
+IMAGES_COLLECTION=u'test3'
+IMAGES_SUB_COLLECTION=u'pipelineRun'
+
+# pylint: disable=protected-access,attribute-defined-outside-init,arguments-differ,abstract-method
+def initialize_database():
+    """
+    Initializes the project's database for writing/reading/updating/deleting purposes.
+    """
+    if not firebase_admin._apps:
+        firebase_admin.initialize_app(credentials.ApplicationDefault(), {
+        'projectId': 'step-project-ellispis',
+        })
+    return firestore.client()
+
+class UploadToDatabase(beam.DoFn):
+    """
+    Uploads asynchronously the information that was extracted from each
```

Comment:
nit: this function knows nothing about 'extraction'. only thing it does is that it store an entity of ImageAttributes type to the DB.